File: 백준/Silver/11004. K번째 수/K번째 수.py
import sys
input = sys.stdin.readline
from heapq import heappop, heappush

N, K = map(int, input().split())
nums = list(map(int, input().split()))
Q = []

for n in nums:
    heappush(Q, n)

for _ in range(K):
    a = heappop(Q)

print(a)

File: 백준/Silver/11004. K번째 수/K번째 수.js
function solution(input) {
  const K = +input[0].split(" ")[1];
  const nums = input[1].split(" ").map((v) => +v);
  const sortedNums = nums.sort((a, b) => a - b);
  return [sortedNums[+K - 1]];
}

(function main() {
  let isLocal = false;
  let fs = require("fs");
  let filePath = isLocal ? "t.txt" : "/dev/stdin";
  let input = fs.readFileSync(filePath).toString().trim().split("\n");

  const answer = solution(input).join("\n");
  console.log(answer);
})();
